scipy/stats/tests/test_distributions.py

```diff
@@ -2325,6 +2325,69 @@ def test_isf(self):
         x = stats.laplace.isf(p)
         assert_allclose(x, -np.log(2*p), rtol=1e-13)
 
+class TestPowerlaw(object):
+    @pytest.mark.parametrize('a', [.5, 1.5, 5])
+    def test_fit_shape(self, a):
+        data = stats.powerlaw.rvs(size=1000, a=a)
+        # obtain the unpenalized log-likelihood function
+
+        def ll(args, data):
+            # to get the same behavior as `powerlaw.reduce_func`, negate the
+            # log-likelihood function
+            return -np.sum(np.log(stats.powerlaw.pdf(data, *args)), axis=0)
+
+        # compare the results of the LL
+        # a warning is raised when data is generated in some situations, so
+        # it will be supressed.
+        with np.errstate(divide='ignore'):
+            _assert_less_or_close_loglike(stats.powerlaw, data, ll)
+
+    @pytest.fixture(scope='function')
+    def rng(self):
+        return np.random.default_rng(1234)
+
+    # @pytest.mark.filterwarnings("ignore:invalid value encountered in ")
+    @pytest.mark.parametrize("rvs_shape", [.1, .5, .75, 1, 2])
+    @pytest.mark.parametrize("rvs_loc", [-1, 0, 1])
+    @pytest.mark.parametrize("rvs_scale", [.1, 1, 5])
+    @pytest.mark.parametrize('fix_shape, fix_loc, fix_scale',
+                             # [[False, False, False]])
+                             [p for p in product([True, False], repeat=3)
+                              if False in p])
+    def test_fit_MLE_comp_optimzer(self, rvs_shape, rvs_loc, rvs_scale,
+                                   fix_shape, fix_loc, fix_scale, rng):
+        print(rvs_shape, rvs_loc, rvs_scale,
+                                   fix_shape, fix_loc, fix_scale)
+
+        data = stats.powerlaw.rvs(size=250, a=rvs_shape, loc=rvs_loc, scale=rvs_scale,
+                                  random_state=rng)
+
+        args = [data, (stats.powerlaw._fitstart(data), )]
+        func = stats.powerlaw._reduce_func(args, {})[1]
+
+        kwds = dict()
+        if fix_shape:
+            kwds['f0'] = rvs_shape
+        if fix_loc:
+            kwds['floc'] = np.nextafter(data.min(), -np.inf)
+        if fix_scale:
+            kwds['fscale'] = rvs_scale
+
+        _assert_less_or_close_loglike(stats.powerlaw, data, func, **kwds)
+
+    def test_fit_warnings(self):
+        assert_fit_warnings(stats.powerlaw)
+        # test for error when `fscale + floc <= np.max(data)` is not satisfied
+        with assert_raises(FitDataError) as e:
+            stats.powerlaw.fit([1, 2, 4], floc=0, fscale=3)
+        assert r" Maximum likelihood estimation with 'powerlaw' requires" in str(e.value)
+
+        # test for error when `data - floc >= 0`  is not satisfied
+        with assert_raises(ValueError) as e:
+            stats.powerlaw.fit([1, 2, 4], floc=2)
+        assert r" Maximum likelihood estimation with 'powerlaw' requires" in str(e.value)
+
+
```

Comment:
```suggestion
```
PEP8 too many blank lines

Reply:
Also, I just noticed that the test `test_fit_MLE_comp_optimzer` is misspelled in four places in the test file. I will add the `i` in optimizer that in this PR.